linode/instance/helpers.go

```diff
@@ -1368,3 +1369,137 @@ func instanceIPSliceToString(ips []*linodego.InstanceIP) []string {
 
 	return result
 }
+
+// Return whether a VPC interface is included in either slice
+func VPCInterfaceIncluded(
+	interfaces1 []linodego.InstanceConfigInterface,
+	interfaces2 []linodego.InstanceConfigInterfaceCreateOptions,
+) (included bool) {
+	for _, ni := range interfaces1 {
+		included = included || ni.Purpose == linodego.InterfacePurposeVPC
+	}
+
+	for _, ni := range interfaces2 {
+		included = included || ni.Purpose == linodego.InterfacePurposeVPC
+	}
+
+	return included
+}
```

Comment:
Optional micro-optimization: 

```suggestion
// VPCInterfaceIncluded returns whether a VPC interface is included in either slice
func VPCInterfaceIncluded(
	interfaces1 []linodego.InstanceConfigInterface,
	interfaces2 []linodego.InstanceConfigInterfaceCreateOptions,
) bool {
	for _, ni := range interfaces1 {
		if ni.Purpose == linodego.InterfacePurposeVPC {
			return true
		}
	}

	for _, ni := range interfaces2 {
		if ni.Purpose == linodego.InterfacePurposeVPC {
			return true
		}
	}

	return false
}
```